### PR TITLE
Change remote-client API function names #155

### LINKED ISSUE #155: Change remote-client API function names
As per feedback, we will change the remote-client API function names to be simpler and follow more standard naming conventions:

```
register_passive_user               -> register_user
create_passive_digital_asset_key    -> create_key
request_transaction_signature       -> sign_transaction
import_asset_key                    -> import_key (or make key material an optional argument to create_key)
export_asset_key                    -> export_key
retrieve_public_keys                -> get_keys
retrieve_public_key_by_id           -> get_key_by_id
retrieve_audit_log                  -> get_log
```

### PR DESCRIPTION
**Pull request by vidyaap**
Closes #155

### CODE REVIEW

File: dams-remote-client/src/api.rs [hunk 1]
```diff
@@ -144,7 +144,7 @@ pub fn retrieve_public_keys(user_id: UserId) -> Result<Vec<KeyInfo>, Error> {
 ///
 /// TODO #30 (design, implementation): Pass a session.
 #[allow(unused)]
-pub fn retrieve_public_key_by_id(user_id: UserId, key_id: &KeyId) -> Result<KeyInfo, Error> {
+pub fn get_key_by_id(user_id: UserId, key_id: &KeyId) -> Result<KeyInfo, Error> {
```
Comment (indomitableSwan) [inline dams-remote-client/src/api.rs:147]: Same comment:  Maybe`get_pub_key_by_id` here?

**Review by indomitableSwan — COMMENTED** [review]
Looks good, except for I do have concerns about the semantics of `get_keys` and `get_key_by_id`, because that sounds like you might be retrieving secret key material, which isn't the case here. What are your thoughts on this?

**Comment by vidyaap** [comment]
> Looks good, except for I do have concerns about the semantics of `get_keys` and `get_key_by_id`, because that sounds like you might be retrieving secret key material, which isn't the case here. What are your thoughts on this?

I agree, I like specifying that it's the public material. Added a commit for that.

**Review by indomitableSwan — APPROVED** [review]
LGTM. Thanks, Vidya!

**Pull request merged**